amaranth/hdl/_dsl.py

```diff
@@ -372,14 +372,20 @@ def Default(self):
             self._statements = _outer_case
 
     @contextmanager
-    def FSM(self, reset=None, domain="sync", name="fsm"):
+    def FSM(self, init=None, reset=None, domain="sync", name="fsm"):
```

Comment:
I fucked up in not forbidding positionals here, so it has to be like this or it breaks the interface:

```suggestion
    def FSM(self, init=None, domain="sync", name="fsm", *, reset=None):
```